### PR TITLE
Andrew.ardito/versiontracking

### PR DESCRIPTION
**Pull request by andrewardito**
### What does this PR do?
Add Version Tracking apm page

### CODE REVIEW

File: content/en/tracing/version_tracking/_index.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+Rolling deploys are a common way to do zero-downtime deployments.  The new version is replicated to all existing hosts or containers one-by-one, while traffic is directed to the other instances.
+
+Using Datadog, you can monitor your rolling deployments of new versions and well as quickly detect any error increases.
+
+{{< img src="tracing/version_tracking/rolling.png" alt="Versions on the Service Page"  style="width:100%;">}}
+
+### Blue/Green Deploys
+
+Blue/Green or other color combination deployments are a strategy to reduce downtime.  Depending on the particular configuration in your environments, this can either mean you are running two clusters of services that are both accepting traffic, or one is on standby, ready to be activated if there are problems with the other.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:31]:
```suggestion
Blue/green (or other color combination) deployments reduce downtime by running two clusters of services that are both accepting traffic, or by keeping one on standby, ready to be activated if there are problems with the other.
```

File: content/en/tracing/version_tracking/_index.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+In either case, setting and viewing the version tags for these services in Datadog allows the comparison requests and errors to detect if one of your clusters is having problems, whether that be an error rate higher than the other or than you SLOs, or if a cluster that is not supposed to be receiving traffic is.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:33]:
```suggestion
Setting and viewing the version tags for these services lets you compare requests and errors to detect if one of the clusters has error rate higher than the other cluster, if a cluster is not meeting SLOs, or if a cluster that is not supposed to be receiving traffic is.
```

File: content/en/tracing/version_tracking/_index.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+### Canary Deploys
+
+Canary deploys allow for a service to be deployed on a limited number of hosts or for a fraction of customers to test a new deployment with limited impact.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:37]:
```suggestion
With canary deploys, a service is deployed on a limited number of hosts or for a fraction of customers, to test a new deployment with limited impact.
```

File: content/en/tracing/version_tracking/_index.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+{{< img src="tracing/version_tracking/rolling.png" alt="Versions on the Service Page"  style="width:100%;">}}
+
+### Blue/Green Deploys
+
+Blue/Green or other color combination deployments are a strategy to reduce downtime.  Depending on the particular configuration in your environments, this can either mean you are running two clusters of services that are both accepting traffic, or one is on standby, ready to be activated if there are problems with the other.
+
+In either case, setting and viewing the version tags for these services in Datadog allows the comparison requests and errors to detect if one of your clusters is having problems, whether that be an error rate higher than the other or than you SLOs, or if a cluster that is not supposed to be receiving traffic is.
+
+### Canary Deploys
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:35]:
```suggestion
### Canary deploys
```

File: content/en/tracing/version_tracking/_index.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+Canary deploys allow for a service to be deployed on a limited number of hosts or for a fraction of customers to test a new deployment with limited impact.
+
+Using Version tags within Datadog allows you to compare error rates, traces, and service behavior for the canary deployment.
+
+For example, you can see in the below image that a canary version was deployed, had a few errors, and was removed afterwards, with traces corresponding to that version available for investigation without any further impact to webservice integrity.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:41]:
```suggestion
For example, you can see in the following image that a canary version was deployed, had a few errors, and was removed, with traces corresponding to that version available for investigation without any further impact.
```

File: content/en/tracing/version_tracking/_index.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+{{< img src="tracing/version_tracking/Canary.png" alt="Versions on the Service Page"  style="width:100%;">}}
+
+### Shadow Deploys
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:45]:
```suggestion
### Shadow deploys
```

File: content/en/tracing/version_tracking/_index.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+Using a Shadow Deployment strategy, a release candidate version is deployed to production to "shadow" the existing deployed service.   Incoming traffic is sent to both services, without impacting production users.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:47]:
```suggestion
In a shadow deployment, a release candidate version is deployed alongside the production version, and incoming traffic is sent to both services, with users seeing the results only from production, but letting you collect data from both.
```

File: content/en/tracing/version_tracking/_index.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+Doing this allows testing a potential version against real production traffic.  Tagging these shadow deployments with a version tag within Datadog allows you to compare error rates, traces, and service behavior between the two versions to determine if the shadow version should be released.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:49]:
```suggestion
Shadow deploys allow you to test a potential release against real production traffic. Tagging shadows with a version tag lets you to compare error rates, traces, and service behavior between the two versions to determine if the shadow version should be released.
```

File: content/en/tracing/version_tracking/_index.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+## Using Version Tags within Datadog
+
+Version tags can be used anywhere within Datadog, whether to filter a search view to a specific version, or compare versions.
+
+### The Service and Resource Page
+
+When version tags are present, the Service and Resource pages have te
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:57]: mporarily left for the weekend 🤣

File: content/en/tracing/version_tracking/_index.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+### Shadow Deploys
+
+Using a Shadow Deployment strategy, a release candidate version is deployed to production to "shadow" the existing deployed service.   Incoming traffic is sent to both services, without impacting production users.
+
+Doing this allows testing a potential version against real production traffic.  Tagging these shadow deployments with a version tag within Datadog allows you to compare error rates, traces, and service behavior between the two versions to determine if the shadow version should be released.
+
+## Using Version Tags within Datadog
+
+Version tags can be used anywhere within Datadog, whether to filter a search view to a specific version, or compare versions.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:53]:
```suggestion
Version tags can be used anywhere within Datadog, whether to filter a search view to a specific version, or to compare metrics from different versions.
```

File: content/en/tracing/version_tracking/_index.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+### The Service and Resource Page
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:55]:
```suggestion
### The Service and Resource pages
```

File: content/en/tracing/version_tracking/_index.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+When version tags are present, the Service and Resource pages have te
+
+#### Service Page
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:59]: Too much heading-nesting. Let's just delete lines 55-57, and make line 59 and 75 an H3

File: content/en/tracing/version_tracking/_index.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+{{< img src="tracing/version_tracking/ServicePage.png" alt="Versions on the Service Page"  style="width:100%;">}}
+
+On the service page, if the version tag is available, the requests widget can be scoped to either of:
+- Total Requests by Version
+- Requests per second by Version
+
+While the errors widget can be scoped to one of three options including version:
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:67]:
```suggestion
The errors widget can be scoped to one of three options that involve Version:
```

File: content/en/tracing/version_tracking/_index.md (hunk 12 above)
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:59]:
```suggestion
### Service page
```

File: content/en/tracing/version_tracking/_index.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+- Total Errors by Version
+- Errors per second by Version
+- % Error rate by Version
+
+All of these can be exported to dashboards and monitors.
+
+#### Resource Page
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:75]:
```suggestion
### Resource page
```

File: content/en/tracing/version_tracking/_index.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+
+### The Service and Resource Page
+
+When version tags are present, the Service and Resource pages have te
+
+#### Service Page
+
+{{< img src="tracing/version_tracking/ServicePage.png" alt="Versions on the Service Page"  style="width:100%;">}}
+
+On the service page, if the version tag is available, the requests widget can be scoped to either of:
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:63]:
```suggestion
On the Service page, if the version tag is available, the requests widget can be scoped to either of:
```

File: content/en/tracing/version_tracking/_index.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+- Errors per second by Version
+- % Error rate by Version
+
+All of these can be exported to dashboards and monitors.
+
+#### Resource Page
+
+{{< img src="tracing/version_tracking/ResourcePage.png" alt="Versions on the Resource Page"  style="width:100%;">}}
+
+On the resource page, if the version tag is available, the requests widget can be scoped to either of:
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:79]:
```suggestion
On the Resource page, if the version tag is available, the requests widget can be scoped to either of:
```

File: content/en/tracing/version_tracking/_index.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+- Total Requests by Version
+- Requests per second by Version
+
+While the errors widget can be scoped to one of three options including version:
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:83]:
```suggestion
The errors widget can be scoped to one of three options that involve Version:
```

File: content/en/tracing/version_tracking/_index.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+- Errors per second by Version
+- % Error rate by Version
+
+All of these can be exported to dashboards and monitors.
+
+### Trace Search and Analytics
+
+{{< img src="tracing/version_tracking/AnalyticsErrorsByVersion.gif" alt="Versions on the Resource Page"  style="width:100%;">}}
+
+When available, Version can be used as a tag for both Trace Search and Analytics, either to filter the live search mode, or indexed traces, or to filter or group analytics queries.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:95]:
```suggestion
When available, Version can be used as a tag for both Trace Search and Analytics, either to filter the live search mode and indexed traces, or to filter or group analytics queries.
```

File: content/en/tracing/version_tracking/_index.md [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+
+Analytics including filtering on the version tag can also be exported to dashboards and monitors for alerting purposes.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:97]:
```suggestion
Analytics, including filtering on the Version tag, can be exported to dashboards and monitors.
```

File: content/en/tracing/version_tracking/_index.md [hunk 20]
```diff
@@ -0,0 +1,104 @@
+---
+title: Version Tracking
+kind: documentation
+description: 'Send traces from your serverless functions to Datadog'
```
Comment (andrewardito) [inline content/en/tracing/version_tracking/_index.md:4]:
```suggestion
description: 'Use Datadog to track version during and after deployments'
```

File: content/en/tracing/version_tracking/_index.md (hunk 9 above)
Comment (andrewardito) [reply content/en/tracing/version_tracking/_index.md:57]: Why do I always do this

File: content/en/tracing/version_tracking/_index.md [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+---
+## The Version Tag
+
+`Version` is a [reserved tag][1] within Datadog, and because of Unified Service Tagging, the `Version` tag is applied across infrastructure, traces, trace metrics, profiles, and logs.
+
+This page describes specific ways you can use the version tag to better monitor deployments and service behavior. See the [Unified Service Tagging][1] docs for instructions for setting up the tags.
+
+## Version Deployment Strategies
+
+Datadog's Version Tracking gives you visibility into the performance of deployed code when you are using the following deployment strategies or others to detect bad code deployments, contain the impact of changes and respond faster to incidents.
```
Comment (kayayarai) [inline content/en/tracing/version_tracking/_index.md:19]:
```suggestion
Datadog's Version Tracking gives you visibility into the performance of deployed code when you are using the following deployment strategies (or others) to detect bad code deployments, contain the impact of changes, and respond faster to incidents.
```

**Pull request merged**